shopify/collection.py

```diff
@@ -0,0 +1,173 @@
+
+        self._next = None
+        self._previous = None
+        self._current_iter = None
+
+        self._no_iter_next = kwargs.pop("no_iter_next", False)
+
+    def _check_pagination_metadata(self):
+        if not ("pagination" in self.metadata
+                and "resource_class" in self.metadata):
+            raise AttributeError("Cursor-based pagination requires "
+                                 "\"pagination\" and \"resource_class\" "
+                                 "attributes in the metadata.")
+
+    def _get_query_params(self, url):
+        """Returns a dictionary of query arguments for the given URL."""
+        parsed_url = urlparse(url)
+        query_dict = parse_qs(parsed_url.query)
+
+        # pyactiveresource.util.to_query considers list items to be multiple
+        # query items and so appends a [] to them which is wrong, because it
+        # fails to consider lists with a length of 1 (which is always returned
+        # from parse_qs). We need to flatten the items.
+        return {
+            k: v[0] for k, v in query_dict.items()
+        }
+
+    def previous(self, no_cache=False, **extra_params):
+        """Returns the previous page of items.
+
+        Args:
+            no_cache: If true the page will not be cached.
+            extra_params: Any extra parameters passed to find().
+        Returns:
+            A PaginatedCollection object with the new data set.
+        """
+        if self._previous:
+            return self._previous
+
+        self._check_pagination_metadata()
+
+        pagination = self.metadata["pagination"]
+        if not "previous" in pagination:
+            raise IndexError("No previous page")
+
+        Resource = self.metadata["resource_class"]
+        query_params = self._get_query_params(pagination["previous"])
+        query_params.update(extra_params)
+
+        previous = Resource.find(id_=None, from_=None, **query_params)
+        if not no_cache:
+            self._previous = previous
+            self._previous._next = self
+
+        previous._no_iter_next = self._no_iter_next
+        return previous
+
+    def next(self, no_cache=False, **extra_params):
+        """Returns the next page of items.
+
+        Args:
+            no_cache: If true the page will not be cached.
+            extra_params: Any extra parameters passed to find().
+        Returns:
+            A PaginatedCollection object with the new data set.
+        """
+        if self._next:
+            return self._next
+
+        self._check_pagination_metadata()
+
+        pagination = self.metadata["pagination"]
+        if not "next" in pagination:
+            raise IndexError("No next page")
+
+        Resource = self.metadata["resource_class"]
+        query_params = self._get_query_params(pagination["next"])
+        query_params.update(extra_params)
+
+        next = Resource.find(id_=None, from_=None, **query_params)
```

Comment:
Looking great, thanks so much!

Does this support the case where we have params in the URL, not just querystring? For example: `collection_listings/<collection id>/product_ids.json`

This was caught and fixed in our Ruby gem for this feature [here](https://github.com/Shopify/shopify_api/pull/611/files#diff-c2896f1ae7d99a3a14ebe68df60ad0c7), and it relies on simply passing the next/prev links into the `from` param, which it looks like pyresource does have, which is good.

It does mean that'd you'd probably (?) have to drop the ability to pass in variables to the `extra_params` in your next/previous methods here. IMO that's fine, as I don't actually see that many valid use cases for wanting to find next/previous pages with new param. You also could then drop the params parsing as well.